cdcsdk-testing/src/test/java/com/yugabyte/cdcsdk/testing/KinesisConsumerIT.java

```diff
@@ -0,0 +1,177 @@
+                "{\"id\":0,\"days_worked\":25.0,\"__deleted\":\"false\"}",
+                "{\"id\":0,\"__deleted\":\"true\"}");
+
+        List<String> actual_data = new ArrayList<>();
+
+        AWSCredentials awsCredentials = new BasicSessionCredentials(System.getenv("AWS_ACCESS_KEY_ID"),
+                System.getenv("AWS_SECRET_ACCESS_KEY"),
+                System.getenv("AWS_SESSION_TOKEN"));
+
+        AmazonKinesisClient client = new AmazonKinesisClient(awsCredentials);
+
+        client.setEndpoint("https://kinesis.ap-south-1.amazonaws.com/");
+
+        final int INTERVAL = 2000;
+        ;
+
+        List<Shard> initialShardData = client.describeStream(stream_name).getStreamDescription().getShards();
+
+        // Getting shardIterators (at beginning sequence number) for reach shard
+        List<String> initialShardIterators = initialShardData.stream()
+                .map(s -> client.getShardIterator(new GetShardIteratorRequest()
+                        .withStreamName(stream_name)
+                        .withShardId(s.getShardId())
+                        .withStartingSequenceNumber(s.getSequenceNumberRange().getStartingSequenceNumber())
+                        .withShardIteratorType(ShardIteratorType.AFTER_SEQUENCE_NUMBER))
+                        .getShardIterator())
+                .collect(Collectors.toList());
+
+        String shardIterator = initialShardIterators.get(0);
+
+        // Continuously read data records from a shard
+        while (true) {
+            GetRecordsRequest getRecordsRequest = new GetRecordsRequest();
+            getRecordsRequest.setShardIterator(shardIterator);
+            getRecordsRequest.setLimit(25);
+
+            GetRecordsResult recordResult = client.getRecords(getRecordsRequest);
+
+            recordResult.getRecords().forEach(record -> {
+                try {
+                    String rec = new String(record.getData().array(), "UTF-8");
+                    actual_data.add(rec);
+                }
+                catch (UnsupportedEncodingException e) {
+                    System.out.println("Could not decode message from Kinesis stream result");
+                    e.printStackTrace();
+                }
+            });
+
+            if (recordResult.getRecords().size() == 0)
+                break;
+            try {
+                Thread.sleep(INTERVAL);
+            }
+            catch (InterruptedException exception) {
+                System.out.println("Receving InterruptedException. Exiting ...");
+                return;
+            }
+            shardIterator = recordResult.getNextShardIterator();
+        }
+
+        assertTrue(validateRecords(expected_data, actual_data));
+
+    }
+
+    public String getUUID() {
+        UUID uuid = UUID.randomUUID();
+        return uuid.toString();
+    }
+
+    public Boolean validateRecords(List<String> expectedData, List<String> actualData) {
```

Comment:
Optional: Is it possible to standardise this function across all tests and move to the base class ?

Reply:
CdcsdkTestBase class did not have any similar functions. So, I haven't moved it there. Should I move it in this class or some other class?

Reply:
Yes. Move it to `CdcsdkTestBase` and reuse it in pub/sub atleast to begin with.

Reply:
Done